### ISSUE TITLE
Crashes on 2008 R2

### ISSUE THREAD

**Issue by thoneu** [Microsoft/ChakraCore#5595]
ChakraCore 1.10.1 does not work anymore on Windows Server 2008 R2. The server is fully patched. Just try to run the ch.exe running a simple script and it will crash.

ChakraCore 1.10.0 works fine.

**Comment by MSLaguana** [comment]
I just tried this out on a Windows Server 2008 R2 SP1 machine and I didn't see a crash with a simple `console.log("Hello world")` test script. Did you try it on an SP1 machine, or just plain 2008 R2 without SP1?

**Comment by thoneu** [comment]
Here are some more details:

It is a Windows Server 2008 R2 Enterprise with SP1 running 64-bit. It is used as a build server, so IIS is installed, Java and Jenkins. Also SQL Server 2008 R2 is installed.

ch.exe run in a command prompt started with administrator rights.

Here is what Windows reports when running ch.exe:

Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	ch.exe
  Application Version:	1.10.1.0
  Application Timestamp:	5b450862
  Fault Module Name:	chakracore.dll
  Fault Module Version:	1.10.1.0
  Fault Module Timestamp:	5b450802
  Exception Code:	80004005
  Exception Offset:	00000000001d8aae
  OS Version:	6.1.7601.2.1.0.274.10
  Locale ID:	1030
  Additional Information 1:	f247
  Additional Information 2:	f247aef3dd090caa0c25df9db7e73353
  Additional Information 3:	1da6
  Additional Information 4:	1da6ec932e37c50c0d96228219b6069e

Read our privacy statement online:
  http://go.microsoft.com/fwlink/?linkid=104288&clcid=0x0409

If the online privacy statement is not available, please read our privacy statement offline:
  C:\Windows\system32\en-US\erofflps.txt

**Comment by thoneu** [comment]
I have just tried to make a fresh install of Windows Server 2008 R2 Enterprise with SP1 on a virtual machine (VirtualBox). Taken the image from MSDN. No updates are made. It still crash with same error.

Can it have something to do when running in a virtual machine? Our build server is also a virtual machine.

**Comment by MSLaguana** [comment]
Thanks, that was exactly what I was after.

It looks like we are failing to initialize winglob which we use for classifying Unicode characters. Does your VM have IE 11 installed on it? There is a dll that gets installed with ie11 that we use for globalization support, and we may be failing to find that.

I am surprised that this only started happening with 1.10.1 though, we've had this dependency for a long time I thought. @jackhorton are you aware of any recent changes to the way we depend on winglob/icu (specifically winglob in this case)? This is the stack that I'm seeing

```
01 00000000`01f3f0c0 000007fe`f4f88aae chakracore!ReportFatalException+0x51 [e:\a\_work\29\s\lib\common\exceptions\reporterror.cpp @ 33] 
02 00000000`01f3f110 000007fe`f4ec3d37 chakracore!<lambda_00a50ea8e5614fd62a6238c2fac4bd85>::operator()+0x6e [e:\a\_work\29\s\lib\runtime\platformagnostic\platform\windows\unicodetext.cpp @ 420] 
03 (Inline Function) --------`-------- chakracore!PlatformAgnostic::UnicodeText::ExecuteWithThreadContext+0x12 [e:\a\_work\29\s\lib\runtime\platformagnostic\platform\windows\unicodetext.cpp @ 182] 
04 (Inline Function) --------`-------- chakracore!PlatformAgnostic::UnicodeText::IsExternalUnicodeLibraryAvailable+0x12 [e:\a\_work\29\s\lib\runtime\platformagnostic\platform\windows\unicodetext.cpp @ 398] 
05 00000000`01f3f140 000007fe`f4e98a40 chakracore!Js::CharClassifier::CharClassifier+0x1b [e:\a\_work\29\s\lib\parser\charclassifier.cpp @ 416] 
06 00000000`01f3f170 000007fe`f4f77633 chakracore!Js::ScriptContext::InitializeAllocations+0x3c [e:\a\_work\29\s\lib\runtime\base\scriptcontext.cpp @ 336] 
```

**Comment by MSLaguana** [comment]
I spoke with @jackhorton and it turns out this is something that had been reported elsewhere, and should also be fixed in 1.10.2 which went out yesterday. Prior to 1.10.1 we didn't crash if we couldn't access the globalization libraries, but we did potentially have wrong behavior. In most cases this doesn't have an impact, but if you have a script with non-ascii Unicode characters in some locations they would be mis-attributed. 1.10.1 turned this into an early crash, but 1.10.2 goes back to silently continuing on with potentially wrong behavior.

tl;dr: You should be able to work around this by updating to 1.10.2, or you can fix it properly by ensuring that the globalization library is available; on win7 era systems like server 2008, the simplest way is to install IE11.

**Comment by thoneu** [comment]
1.10.2 works fine. After installing IE11, 1.10.1 works again. Thanks.
